clients/vscode-hlasmplugin/src/commentEditorCommands.ts

```diff
@@ -0,0 +1,186 @@
+    return text.length > 71 && text.substring(71, 72) !== ' ';
+}
+
+function findFirstLine(doc: vscode.TextDocument, lineno: number): number {
+    while (lineno > 0 && isContinued(doc.lineAt(lineno - 1).text))
+        --lineno;
+
+    return lineno;
+}
+
+function findLastLine(doc: vscode.TextDocument, lineno: number): number {
+    while (lineno < doc.lineCount && isContinued(doc.lineAt(lineno).text))
+        ++lineno;
+
+    return lineno;
+}
+
+function isCommented(doc: vscode.TextDocument, lineno: number): boolean {
+    const text = doc.lineAt(lineno).text;
+    return text.startsWith('*') || text.startsWith('.*');
+}
+
+function addComment(editor: vscode.TextEditor, edit: vscode.TextEditorEdit, lineno: number) {
+    if (editor.document.lineAt(lineno).text.length > 0)
+        edit.replace(new vscode.Selection(new vscode.Position(lineno, 0), new vscode.Position(lineno, 1)), '*');
+    else
+        edit.insert(new vscode.Position(lineno, 0), '*');
+}
+
+function removeComment(editor: vscode.TextEditor, edit: vscode.TextEditorEdit, lineno: number) {
+    const text = editor.document.lineAt(lineno).text;
+    let replacement = ' ';
+    let i = 1;
+    for (; i < text.length; ++i) {
+        if (text.charAt(i) !== '*') {
+            replacement = text.charAt(i);
+            break;
+        }
+    }
+    replacement = replacement.repeat(i);
+
+    edit.replace(new vscode.Selection(new vscode.Position(lineno, 0), new vscode.Position(lineno, i)), replacement);
+}
+
+export function lineCommentCommand(editor: vscode.TextEditor, edit: vscode.TextEditorEdit, args: CommentOption) {
+    if (editor.document.isClosed) return;
+
+    const lines = [...new Set(editor.selections.flatMap(x => seq(x.start.line, x.end.line)).map(x => findFirstLine(editor.document, x)))].sort((l, r) => l - r);
+    const lineWithStatus = lines.map(x => { return { lineno: x, commented: isCommented(editor.document, x) }; })
+
+    for (const { lineno, commented } of lineWithStatus) {
+        switch (args) {
+            case CommentOption.toggle:
+                if (commented)
+                    removeComment(editor, edit, lineno);
+                else
+                    addComment(editor, edit, lineno);
+                break;
+
+            case CommentOption.add:
+                if (!commented)
+                    addComment(editor, edit, lineno);
+                break;
+
+            case CommentOption.remove:
+                if (commented)
+                    removeComment(editor, edit, lineno);
+                break;
+        }
+    }
+}
+
+interface CodeBlock {
+    first: number;
+    last: number;
+}
+
+function isolateBlocks(block_candidates: CodeBlock[]): CodeBlock[] {
+    let last_last = -1;
+    const blocks: CodeBlock[] = [];
+
+    for (const b of block_candidates) {
+        if (b.first > last_last)
+            blocks.push(b);
+        else
+            blocks[blocks.length - 1].last = Math.max(b.last, last_last);

```

Comment:
Array.prototype.at() can also be used if you find it better:
```suggestion
            blocks.at(-1).last = Math.max(b.last, last_last);
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at